connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -47,11 +52,37 @@ private[client] object GrpcExceptionConverter {
     }
   }
 
-  private def toSparkThrowable(ex: StatusRuntimeException): SparkThrowable with Throwable = {
+  private val errorFactory = new ErrorFactoryBuilder()
+    .addConstructor((message, _) => new ParseException(None, message, Origin(), Origin()))
+    .addConstructor((message, _) => new AnalysisException(message))
+    .build()
+
+  private def errorInfoToThrowable(info: ErrorInfo, message: String): Throwable = {
+    val classes =
+      mapper.readValue(info.getMetadataOrDefault("classes", "[]"), classOf[Array[String]])
+
+    for (cls <- classes) {
+      errorFactory.get(cls) match {
+        case Some(constructor) =>
+          return constructor(message, null)
+        case None =>
+        // Do nothing
+      }
+    }
+
+    new SparkException(message)
```

Comment:
Why don't you return `Option[Throwable]` instead of throwing an exception here? I think it's easier to follow the code if there's only one place that throws an exception.